### ISSUE TITLE
Update Vertical Navigation to be more scalable

### ISSUE THREAD

**Issue by Rohoover** [patternfly/patternfly-design#1027]
Currently the vertical navigation continually adds left-hand nav items that expand. For some applications, the length becomes quite long. The work-around of using secondary tabs in the content area is not idea since applications are already using one to two layers of tabs + a context selector in the header content area. Ideally, we would reintroduce sub menus similar to what was available for the PF3 navigation.

This currently affects Satellite and Insights. Satellite is requesting a new design for the 7.0 release (with development resources to contribute) and Insights is now pushing towards expansion post Summit that would benefit from the new design.

**Comment by trahman73** [comment]
The Ansible Automation Bundle is also limited in it's navigation choices by this in both our on-prem and cloud components. We are currently having to use a lot of sub tabs within the page as a means to get around this issue.

**Comment by KendraMar** [comment]
Yeah, hearing from Taufique and the Ansible folks about this as well.

The team favors the flyout approach that was available in PatternFly 3...we'd like to explore that.

**Comment by mcarrano** [comment]
I started a discovery document here to gather some requirements and prior work: https://docs.google.com/document/d/1Q5_KVOpA1IwRO9VCNfn9k4BqPfrhLXeL7JtJPYAWOeU/edit?usp=sharing

**Comment by mcarrano** [comment]
@beccaglass Per our discussion today, I'm reassigning this to you with the assumption that @mceledonia will work closely with you to consult on visuals.

**Comment by MariSvirik** [comment]
**More information about Satellite**
Satellite devs are willing to help with development. But it would be better to have someone from the PF4 team help them with CSS and PF4 classes, so there is no need to rebuild the component later.

**Dates**
To get it in the Satellite 7.0 release, we need designs by August at the latest, so devs have until November (dev freeze) to work on it.

**Comment by nlcwong** [comment]
Cost Management for OpenShift used to be a two level navigation app, but when it needed to move over underneath the OpenShift pillar in cloud.redhat.com, Cost Management itself now became the first level nav option.  We had a need for the nav to support at least three levels so that cost management could maintain its original navigation flow.  Currently, we had to flatten our nav, but would like to have a three level nav available as soon as humanly possible. We cannot provide a sectional landing for the public cloud details, so it makes sense for our use case that when a user expands a section, there are no consolidated landing page for those clouds to then make a choice for (so we'll need something where when a user clicks to expand, they would have to make a choice as to which subset they would need to choose.

Here are the levels (left numbers represent the level)
1 Cost Management
2 Overview
2 OpenShift
2 Public Clouds
3 Amazon Web Services
3 Microsoft Azure
3 Google Cloud Platform
3 IBM Cloud - COMING
3 Alibaba - FUTURE
2 Cost Model
2 Cost Explorer

**Comment by KendraMar** [comment]
Hi, Platform UXD provided 3 design options extending the current Pfly component expand/collapse option. We don't have a cascade/flyout option that we're confident would extend successfully to OpenShift and Satellite. This would need to come from a PatternFly effort.

I believe Satellite and Cost Management UXers have said that they don't need to immediately add third levels of nav to Satellite and cloud.redhat.com and they'd rather wait for a flyout navigation component.

**Comment by beccaglass** [comment]
newest doc:
https://docs.google.com/document/d/17jhcgkUgpw7f9Ixf9ew8RfcivSck9mdv_Yw6fIZhlAs/edit?userstoinvite=rblackbu%40redhat.com&ts=60ddcbea&actionButton=1#heading=h.au2hnxprb8tl

**Comment by mcarrano** [comment]
I've opened https://github.com/patternfly/patternfly/issues/4225 to begin core implementation. 

@beccaglass when done with your final designs, please post in both places and I can mark this as complete. Thanks for all your efforts on this!

**Comment by beccaglass** [comment]
final designs & specs: https://docs.google.com/document/d/17jhcgkUgpw7f9Ixf9ew8RfcivSck9mdv_Yw6fIZhlAs/edit?ts=60cb76dd#heading=h.ea3lujnv46vp

**Comment by mcarrano** [comment]
Thanks, @beccaglass !
